OwnYourData/ClinicalTrials/ClinicalTrialsView.swift

```diff
@@ -73,6 +122,74 @@ struct ClinicalTrialsView: View {
             }
         }
     }
+    
+    // Function to update search parameters
+    private func fetchTrials() async {
+        isLoading = true
+        // Reload trials with updated search parameters
+        trials.removeAll() // Clear existing trials
+        collapseStates.removeAll() // Clear existing collapse states
```

Comment:
A key element that we are missing here is a call to `locationManager.requestWhenInUseAuthorization()` with a mechanism to await location updates. As noted above, I would suggest to use SpeziLocation to abstract this in a nice manner despite it not being fully ready to be released and documented it contains most of the elements you are looking for.